frontend/src/components/Sidebar.tsx

```
@@ -175,6 +236,150 @@ export function Sidebar({
+                className="sr-only"
+                checked={theme === "dark"}
+                onChange={toggleTheme}
+              />
+              <div
+                className={`w-12 h-6 rounded-full shadow-inner transition-colors duration-300 cursor-pointer ${
+                  theme === "dark"
+                    ? "bg-purple-600 dark:bg-purple-600"
+                    : "bg-gray-300 dark:bg-gray-600"
+                }`}
+                onClick={toggleTheme}
+              ></div>
```

Biome (2.1.2)

[error] Static Elements should not be interactive.

To add interactivity such as a mouse or key event listener to a static element, give the element an appropriate role value.

(lint/a11y/noStaticElementInteractions)